code/game/gamemodes/events/powercreeper.dm

```diff
@@ -119,23 +121,50 @@
 
 /obj/structure/cable/powercreeper/proc/isViableGrow(var/turf/T)
 	if(!T.has_gravity())
+		to_chat(world, "[src] tried to cross into [T], but has no gravity")
 		return 0
 	if(!T.Adjacent(src))
+		to_chat(world, "[src] tried to cross into [T], but wasn't adjacent")
 		return 0
-	if(locate(/obj/structure/cable/powercreeper) in T)
+	var/obj/structure/cable/powercreeper/PC = locate(/obj/structure/cable/powercreeper) in T
+	if(PC && PC != src)
+		to_chat(world, "[src] tried to cross into [T], but found a powercreeper there already")
 		return 0
 	if((T.density == 1) || T.has_dense_content())
+		to_chat(world, "[src] tried to cross into [T], but T was dense or had dense content [T.density] [T.has_dense_content()]")
 		return 0
 	return 1
 
 /obj/structure/cable/powercreeper/proc/updateNeighbours(var/dying = FALSE)
 	for(var/dir in cardinal)
-		var/obj/structure/cable/powercreeper/P = locate() in get_step(src, dir)
+		var/turf/T = get_step(src, dir)
+		var/obj/structure/cable/powercreeper/P = locate() in T
 		if(P)
 			if(dying)
 				P.growdirs |= get_dir(P, src)
 			else
 				P.growdirs &= ~get_dir(P, src)
+		var/area/A = get_area(T)
```

Comment:
yeah like i said, why go through the trouble of having an area density change when you can go and subscribe to a tile density change.

additionally to this, this will still not fix moving dense objs eg mechs

Reply:
Still working on that, was just pushing progress so I could see what had been changed better

Reply:
Sorted now.